pkg/install/resource/tm_deployment.go

```diff
@@ -0,0 +1,183 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/datawire/ambassador/pkg/kates"
+	"github.com/datawire/dlib/dlog"
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
+	"github.com/telepresenceio/telepresence/v2/pkg/install"
+)
+
+const managerLicenseName = "systema-license"
+
+type tmDeployment struct {
+	found *kates.Deployment
+}
+
+var TrafficManagerDeployment Instance = &tmDeployment{}
```

Comment:
So this is a global variable with state.  Which isn't a blocker, but... globals are a smell.

Reply:
Maybe we avoid that by having `GetTrafficManagerResources()` say `&tmDeployment{}` instead of `TrafficManagerDeployment`?

Reply:
True that. Bad design. I'll fix it. There's no need for this one to be a variable.